### ISSUE TITLE
paperwork_backend installs egg-info in virtualenv

### ISSUE THREAD

**Issue by gordonmessmer** [fedora-python/pyp2rpm#208]
I've run into an odd situation in that the master branch no longer completes tests normally on my own system, though it seems that it does in Travis CI.

The test output indicates an extra line in the generated spec, compared to the spec in test_data:

```
E         -  u'%{python3_sitelib}/paperwork_backend',
E         ?  -
E         +  '%{python3_sitelib}/paperwork_backend',
E         -  u'%{python3_sitelib}/paperwork_backend-1.3.0-py3.7.egg-info',
E         -  u'%{python3_sitelib}/paperwork_backend-%{version}-py%{python3_version}.egg-info',
E         ?  -
E         +  '%{python3_sitelib}/paperwork_backend-%{version}-py%{python3_version}.egg-info',
```

pyp2rpm.virtualenv.site_packages_filter will remove dist-info and pth from the sitelib directory, and it seems like it probably should remove egg-info as well.  That's a trivial change.  The odd part is that I don't understand why this test doesn't fail in Travis CI's environment.

paperwork-backend does install egg-info when installed with pip in a virtualenv, and I'm unable to find any other package that does that.  I don't see anything in the code that currently strips it out, so I don't know why that line isn't always present.

Do you have any idea why paperwork-backend would install egg-info in a virtualenv's sitelib some of the time, but not consistently?

**Comment by gordonmessmer** [comment]
In a container:

```
[root@92c283052b3c ~]# python3.7 -m venv /tmp/paperwork
[root@92c283052b3c ~]# . /tmp/paperwork/bin/activate
(paperwork) [root@92c283052b3c ~]# pip install paperwork-backend==1.2.4 --no-deps
Collecting paperwork-backend==1.2.4
  Downloading https://files.pythonhosted.org/packages/e6/3c/ba77e8c087e32e2ef1f34b2eff28f7ce0e3f04989e390bfad0a411530345/paperwork-backend-1.2.4.tar.gz (59kB)
Installing collected packages: paperwork-backend
  Running setup.py install for paperwork-backend ... done
Successfully installed paperwork-backend-1.2.4
You are using pip version 19.0.3, however version 19.2.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.

(paperwork) [root@92c283052b3c ~]# ls /tmp/paperwork/lib/python3.7/site-packages/
__pycache__  easy_install.py  paperwork_backend  paperwork_backend-1.2.4-py3.7.egg-info  pip  pip-19.0.3.dist-info  pkg_resources  setuptools  setuptools-40.8.0.dist-info
```

**Comment by gordonmessmer** [comment]
(The egg-info in the test output is version 1.3.0, but that's another bug which we can ignore for now.  The version request isn't currently passed to the metadata extractor, so it downloads the most recent version when building the virtualenv.)

**Comment by hroncok** [comment]
> pyp2rpm.virtualenv.site_packages_filter will remove dist-info and pth from the sitelib directory, and it seems like it probably should remove egg-info as well.

Why? I don't remember what is the purpose here.

**Comment by gordonmessmer** [comment]
If virtualenv-api is available, then pyp2rpm will create a virtualenv and install the python module from pypi in the venv.  It examines the content of the site packages directory to build the rpm files list.  site_packages_filter's doc-comment indicates that it removes wheel distribution information (dist-info and pth).  However, egg-info is left intact, and if that information is installed in the venv, pyp2rpm will build a file list that looks like:

```
%{python3_sitelib}/paperwork_backend
%{python3_sitelib}/paperwork_backend-1.3.0-py3.7.egg-info
%{python3_sitelib}/paperwork_backend-%{version}-py%{python3_version}.egg-info
```

The second line is problematic.  It's redundant, for one.  It also refers to whatever specific version of the module and python version were used in the venv.

Oddly, this behavior is codified in testing.  test_virtualenv.TestUtils.test_site_packages_filter verifies that the egg-info is left intact, so it must be expected, but I cannot explain why.

I get inconsistent results with regard to paperwork-backend.  Outside of testing environment: if I install pyp2rpm with pip, then I don't get virtualenv-api and the problem doesn't manifest because there's no virtualenv extraction support.  If I create a venv and install a specific version (1.2.4) with pip, then I seem to consistently get .egg-info in site-packages.  But if I don't specify a version, I get egg-info in some environments and dist-info in others, and again, I can't explain that at all, right now.

Whatever the cause, I think egg-info should probably be treated the same as dist-info and pth, and filtered out of the site packages.

**Comment by hroncok** [comment]
> Oddly, this behavior is codified in testing. test_virtualenv.TestUtils.test_site_packages_filter verifies that the egg-info is left intact, so it must be expected, but I cannot explain why.

Don't trust that. Often the tests were created to maintain the status quo only.

> Whatever the cause, I think egg-info should probably be treated the same as dist-info and pth, and filtered out of the site packages.

:+1:

**Comment by gordonmessmer** [comment]
PR #207 is running tests now.

**Comment by gordonmessmer** [comment]
Resolved in v3.3.4
